src/panels/lovelace/cards/hui-clock-card.ts

```diff
@@ -0,0 +1,203 @@
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { DateTime, type DateTimeMaybeValid } from "luxon";
+import "../../../components/ha-alert";
+import "../../../components/ha-card";
+import type { HomeAssistant } from "../../../types";
+import type {
+  LovelaceCard,
+  LovelaceCardEditor,
+  LovelaceGridOptions,
+} from "../types";
+import type { ClockCardConfig } from "./types";
+
+@customElement("hui-clock-card")
+export class HuiClockCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import("../editor/config-elements/hui-clock-card-editor");
+    return document.createElement("hui-clock-card-editor");
+  }
+
+  public static getStubConfig(): ClockCardConfig {
+    return {
+      type: "clock",
+      time_format: "hh:mm:ss",
+      clock_size: "medium",
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: ClockCardConfig;
+
+  @state() private _time?: DateTime;
+
+  @state() private _interval = 1000;
+
+  private _tickInterval?: number;
+
+  public setConfig(config: ClockCardConfig): void {
+    if (!config.time_format) {
+      throw new Error("time_format required");
```

Comment:
I dont think it should be required, and it should default to `hass.locale.time_format`